pkg/crc/machine/cert-renewal.go

```diff
@@ -140,105 +41,32 @@ func RegenerateCertificates(sshRunner *ssh.SSHRunner, machineName string) error
 	if startedKubelet {
 		defer sd.Stop("kubelet") //nolint:errcheck
 	}
-	recoveryPod, err := startRecoveryPod(sshRunner)
-	if err != nil {
-		logging.Debugf("Error starting recovery kube-apiserver-operator: %v", err)
-		return err
-	}
-	defer recoveryPod.runPodCommand("recovery-apiserver destroy") //nolint:errcheck
-
-	oc := oc.NewOcConfig(recoveryPod, "admin", "recovery")
-	err = waitForRecoveryPod(oc)
-	if err != nil {
-		logging.Debugf("Error waiting for recovery apiserver to come up: %v", err)
-		return err
-	}
-
-	_, err = recoveryPod.runPodCommand("regenerate-certificates")
-	if err != nil {
-		logging.Debugf("Error running 'regenerate-certificates' command in the kube-apiserver-operator")
-		return err
-	}
-
-	forceRedeploymentPatch := fmt.Sprintf(`'{"spec": {"forceRedeploymentReason": "recovery-%s"}}'`, time.Now().Format(time.RFC3339))
-	for _, resourceType := range []string{"kubeapiserver", "kubecontrollermanager", "kubescheduler"} {
-		_, _, err := oc.RunOcCommand(
-			"patch", resourceType, "cluster",
-			fmt.Sprintf("--patch=%s", forceRedeploymentPatch),
-			"--type=merge")
-		if err != nil {
-			logging.Warnf("Error forcing redeployment of %s: %v", resourceType, err)
-			return err
-		}
-	}
-
-	_, err = sshRunner.Run(fmt.Sprintf("sudo KUBECONFIG=%s /usr/local/bin/recover-kubeconfig.sh >kubeconfig && sudo mv kubeconfig /etc/kubernetes/kubeconfig", recoveryPod.GetKubeconfigPath()))
-	if err != nil {
-		logging.Debugf("Error invoking recover-kubeconfig.sh: %v", err)
-		return err
-	}
-
-	output, _, err := oc.RunOcCommand("get", "configmap", "kube-apiserver-to-kubelet-client-ca",
-		"-n openshift-kube-apiserver-operator",
-		"--template='{{ index .data \"ca-bundle.crt\" }}'")
-	if err != nil {
-		logging.Debugf("Error reading kubelet client CA: %v", err)
-		return err
-	}
-	/* Copy the ca-bundle data we just read to etc/kubernetes/kubelet-ca.crt */
-	err = sshRunner.SetTextContentAsRoot("/etc/kubernetes/kubelet-ca.crt", output, 0644)
-	if err != nil {
-		logging.Debugf("Error writing kubelet client CA to /etc/kubernetes/kubelet-ca.crt: %v", err)
-		return err
-	}
-
-	_, _ = sshRunner.Run("sudo touch /run/machine-config-daemon-force")
-
-	_, err = sd.Stop("kubelet")
-	if err != nil {
-		return err
-	}
-	_, _ = sshRunner.Run("sudo rm -rf /var/lib/kubelet/pki /var/lib/kubelet/kubeconfig")
-	_, _ = sshRunner.Run("sudo crictl stopp $(sudo crictl pods -q)")
-	_, _ = sshRunner.Run("sudo crictl rmp $(sudo crictl pods -q)")
-
-	_, err = sd.Start("kubelet")
-	if err != nil {
-		logging.Debugf("Error restarting kubelet service: %v", err)
-		return err
-	}
-
+	oc := oc.UseOCWithConfig(machineName)
 	/* 2 CSRs to approve, one right after kubelet restart, the other one a few dozen seconds after
-	approving the first one */
+	approving the first one
+	- First one is requested by system:serviceaccount and address to node-bootstrapper
```

Comment:
I don't understand what you mean by "address to xxx"

Reply:
@cfergeau this is more like for me to remember which cert address to which component so when I file an issue I don't need to everytime go back to working cluster and check those details.

Reply:
I don't mind having "node-bootstrapper" and the other one being mentioned, it's the "address to" formulation which I don't understand, so the comment is not as useful as it could be.